### ISSUE TITLE
Equivalent width measurements are different from IRAF

### ISSUE THREAD

**Issue by ganesh1076** [astropy/specutils#841]
Whenever we used specutils to measured EW its measurement is not matching with the measurement from IRAF and the measured value of EW at 5890 A from IRAF is 3.1 A and from specutils is 1.5363474 A.
What could be wrong? Are we missing something in our code?
I am attaching the normalized spectrum for the 5890 \AA and the code below.

import pandas as pd
from astropy import units as u
from specutils.spectra import Spectrum1D, SpectralRegion
from specutils.analysis import equivalent_width
import matplotlib.pyplot as plt

data=pd.read_csv('/home/ganesh/line_5890.txt',header=None)

y=u.Quantity(data[1],u.dimensionless_unscaled)
x=u.Quantity(data[0],u.angstrom)
spectrum = Spectrum1D(spectral_axis=x,flux=y)
Haregion=SpectralRegion(5860*u.angstrom, 5920*u.angstrom)
equivalent_width(spectrum,regions=Haregion)
[line_5890.txt](https://github.com/astropy/specutils/files/6818459/line_5890.txt)

**Comment by dhomeier** [comment]
No error on your side really.
The function is using a simple multiplication of `continuum` with the length of the spectral axis over the boundary point (_centres_), but the actual spectral flux is calculated as in `line_flux`, which is integrating using over `spectral_axis.bin_edges`:
https://github.com/astropy/specutils/blob/1f6eded24d74ca40ec3f370b0b56573f5e314345/specutils/analysis/flux.py#L179-L182
making the continuum spectrum effectively one wavelength step shorter than the flux spectrum. I guess this was missed when `bin_edges` were introduced to represent the actual wavelength range covered by the spectrum. Changing this to
```python
    dx = (np.abs(spectral_axis.bin_edges[-1] - spectral_axis.bin_edges[0]))
```
produces EW = `<Quantity 2.79417127 Angstrom>` for your data; 3.002 Å for a narrower region [5870, 5910] Å.

@nmearl in 843ef92 `_compute_line_flux` was changed to use `bin_edges`, but `_compute_equivalent_width` was not – does this sound correct?

**Comment by ganesh1076** [comment]
Thank you for the clarification, Derek!

**Comment by dhomeier** [comment]
Thanks for bringing this to our attention @ganesh1076!
Do you think the revised value of 2.8 Å makes sense, given that IRAF might be using slightly different methods to determine the continuum level? So that the fix in #843 would bring this to a satisfactory solution?

**Comment by ganesh1076** [comment]
Yes, It is good and that might be due to the noisy data. It was not more than 0.1 for another example :)  you should pull these changes in your main branch.

**Comment by rosteen** [comment]
Sounds like this was addressed well enough by #843, closing.
